core/domain/feedback_jobs_one_off_test.py

```diff
@@ -0,0 +1,89 @@
+
+    def setUp(self):
+        super(PopulateMessageCountOneOffJobTest, self).setUp()
+        feedback_models.GeneralFeedbackThreadModel(
+            id='exp1.thread1', entity_id='exp1', entity_type='state1',
+            original_author_id='author', message_count=0,
+            status=feedback_models.STATUS_CHOICES_OPEN,
+            subject='subject', summary='summary', has_suggestion=False,
+            ).put()
+        feedback_models.GeneralFeedbackMessageModel(
+            id='exp1.thread1.1', thread_id='exp1.thread1', message_id=1,
+            author_id='author', text='message text').put()
+        feedback_models.GeneralFeedbackMessageModel(
+            id='exp1.thread1.2', thread_id='exp1.thread1', message_id=2,
+            author_id='author', text='message text').put()
+        feedback_models.GeneralFeedbackThreadModel(
+            id='exp2.thread2', entity_id='exp2', entity_type='state2',
+            original_author_id='author', message_count=0,
+            status=feedback_models.STATUS_CHOICES_OPEN,
+            subject='subject', summary='summary', has_suggestion=False,
+            ).put()
+        feedback_models.GeneralFeedbackMessageModel(
+            id='exp2.thread2.1', thread_id='exp2.thread2', message_id=1,
+            author_id='author', text='message text').put()
+        feedback_models.GeneralFeedbackThreadModel(
+            id='exp3.thread3', entity_id='exp3', entity_type='state3',
+            original_author_id='author', message_count=0,
+            status=feedback_models.STATUS_CHOICES_OPEN,
+            subject='subject', summary='summary', has_suggestion=False,
+            ).put()
+        feedback_services.set_message_count('exp1.thread1', 0)
+        feedback_services.set_message_count('exp2.thread2', 0)
+        feedback_services.set_message_count('exp3.thread3', 0)
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = (
+            feedback_jobs_one_off
+            .PopulateMessageCountOneOffJob.create_new())
+        feedback_jobs_one_off.PopulateMessageCountOneOffJob.enqueue(
+            job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+
+    def test_message_count_job(self):
+        self.assertEqual(feedback_services.thread_message_count(
+            'exp1.thread1'), 0)
+        self.assertEqual(feedback_services.thread_message_count(
+            'exp2.thread2'), 0)
+        self.assertEqual(feedback_services.thread_message_count(
+            'exp3.thread3'), 0)
+        self._run_one_off_job()
+        self.assertEqual(feedback_services.thread_message_count(
+            'exp1.thread1'), 2)
+        self.assertEqual(feedback_services.thread_message_count(
+            'exp2.thread2'), 1)
+        self.assertEqual(feedback_services.thread_message_count(
+            'exp3.thread3'), 0)
```

Comment:
Is there a way you can test the job output? (If you look at the admin panel there's a way to retrieve the job output, you may be able to use the methods/functions in that pipeline.)

This might help catch issues like the lack of a 2-tuple in the yield statement above.

Reply:
Hey @seanlip, why not add a presubmit check for it in the ```pre_commit_linter.py``` which must ensure that every yield statement in the one-off job should contain a key and a value ?

Reply:
What you suggest ?

Reply:
DONE! Added a check